internal/pagination/books.go

```diff
@@ -0,0 +1,172 @@
+package pagination
+
+import (
+	"bookshop/internal/models"
+	"fmt"
+	"net/http"
+	"strconv"
+	"strings"
+)
+
+const (
+	formatParam   = "format"
+	categoryParam = "category"
+	minPriceParam = "min_price"
+	maxPriceParam = "max_price"
+)
+
+type BookPaginationData struct {
+	*PaginationData
+	Format   []string
+	Category []int64
+	MinPrice float64
+	MaxPrice float64
+}
+
+func ParseBookPaginationQuery(r *http.Request) (*BookPaginationData, error) {
+	qs := r.URL.Query()
+
+	p, err := ParsePaginationData(r, bookItemSortKeyValidator)
+	if err != nil {
+		return nil, err
+	}
+
+	bp := &BookPaginationData{PaginationData: p}
+
+	if formats, err := parseAndValidateFormats(qs.Get(formatParam)); err != nil {
+		return nil, err
+	} else if len(formats) > 0 {
+		bp.Format = formats
+	} else {
+		bp.Format = []string{string(models.Paperback)}
+	}
+
+	if categories, err := parseCategories(qs.Get(categoryParam)); err != nil {
+		return nil, err
+	} else {
+		bp.Category = categories
+	}
+
+	if minPrice, maxPrice, err := parseMinMaxPrice(qs.Get(minPriceParam), qs.Get(maxPriceParam)); err != nil {
+		return nil, err
+	} else {
+		bp.MinPrice = minPrice
+		bp.MaxPrice = maxPrice
+	}
+
+	return bp, nil
+}
+
+func parseAndValidateFormats(formatParam string) ([]string, error) {
+	if formatParam == "" {
+		return nil, nil
+	}
+
+	formats := strings.Split(formatParam, ",")
+	validFormats := []string{}
```

Comment:
Can we declare it like nil slice? var validFormats []string? Generally recommended to declare slice as nil if we don't know it's size. validFormats := []string{} - the empty slice declaration useful only in json-parsing scenarios.

Reply:
Also "validFormats" name is confusing, all these are valid: models.Audiobook, models.EBook, models.Hardcover, models.Paperback